core/Pos/Binary/Class/Core.hs

```diff
@@ -1,171 +1,586 @@
-{-# LANGUAGE CPP #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE PolyKinds           #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies        #-}
+{-# LANGUAGE TypeOperators       #-}
+module Pos.Binary.Class.Core
+    ( Bi(..)
+    , encodeBinary
+    , decodeBinary
+    , enforceSize
+    , matchSize
+    -- * CBOR re-exports
+    , encodeListLen
+    , decodeListLen
+    , decodeListLenOf
+    , Encoding
+    , Decoder
+    -- * GHC-Generics-based encoding & decoding
+    , genericEncode
+    , genericDecode
+    ) where
 
-#include "MachDeps.h"
+import           Codec.CBOR.Decoding
+import           Codec.CBOR.Encoding
+import           GHC.Generics
```

Comment:
Hey @gromakovsky! The `Pos.Binary.Class.Core` module was written almost entirely by @arybczak , which is currently on annual leave for 2 weeks. Whereas I cannot comment directly on naming choices (e.g. `defaultEncodeList` vs `encodeList`) I can certainly amend this PR to match the project's style guide 😉